### PR TITLE
Fix build

### PR DESCRIPTION
**Pull request by TBG-FR**
Hello there ! A small PR that fixes all the issues that I've had when trying to build this project 😉 

### Removed deprecated `dotenv`
```js
npm WARN deprecated @types/dotenv@8.2.0: This is a stub types definition. dotenv provides its own type definitions, so you do not need this installed.
```

### Issues with `@opentelemetry` packages versions
```js
openTelemetry/open-telemetry-tracer.ts:55:5 - error TS2322: Type 'import("/usr/src/app/src/building-blocks/node_modules/@opentelemetry/sdk-trace-base/build/src/SpanProcessor").SpanProcessor' is not assignable to type 'import("/usr/src/app/src/building-blocks/node_modules/@opentelemetry/sdk-node/node_modules/@opentelemetry/sdk-trace-base/build/src/SpanProcessor").SpanProcessor'.
  Types of property 'onStart' are incompatible.
    Type '(span: import("/usr/src/app/src/building-blocks/node_modules/@opentelemetry/sdk-trace-base/build/src/Span").Span, parentContext: import("/usr/src/app/src/building-blocks/node_modules/@opentelemetry/api/build/src/context/types").Context) => void' is not assignable to type '(span: import("/usr/src/app/src/building-blocks/node_modules/@opentelemetry/sdk-node/node_modules/@opentelemetry/sdk-trace-base/build/src/Span").Span, parentContext: import("/usr/src/app/src/building-blocks/node_modules/@opentelemetry/api/build/src/context/types").Context) => void'.
      Types of parameters 'span' and 'span' are incompatible.
        Type 'import("/usr/src/app/src/building-blocks/node_modules/@opentelemetry/sdk-node/node_modules/@opentelemetry/sdk-trace-base/build/src/Span").Span' is not assignable to type 'import("/usr/src/app/src/building-blocks/node_modules/@opentelemetry/sdk-trace-base/build/src/Span").Span'.
          Types have separate declarations of a private property '_spanContext'.

55     spanProcessor: provider.activeSpanProcessor,
       ~~~~~~~~~~~~~

  node_modules/@opentelemetry/sdk-node/build/src/types.d.ts:20:5
    20     spanProcessor: SpanProcessor;
           ~~~~~~~~~~~~~
    The expected type comes from property 'spanProcessor' which is declared here on type 'Partial<NodeSDKConfiguration>'


Found 1 error in openTelemetry/open-telemetry-tracer.ts:55
```
```js
npm WARN deprecated @opentelemetry/node@0.24.0: Package renamed to @opentelemetry/sdk-trace-node
npm WARN deprecated @opentelemetry/api-metrics@0.24.0: Please use @opentelemetry/api >= 1.3.0
npm WARN deprecated @opentelemetry/metrics@0.24.0: Package renamed to @opentelemetry/sdk-metrics-base
npm WARN deprecated @opentelemetry/tracing@0.24.0: Package renamed to @opentelemetry/sdk-trace-base
```

### Missing `OpenTelemetryOptions` export 
```js
bmn-passenger  | Error: Nest can't resolve dependencies of the OpenTelemetryTracer (?). Please make sure that the argument OpenTelemetryOptions at index [0] is available in the PassengerModule context.
bmn-passenger  |
bmn-passenger  | Potential solutions:
bmn-passenger  | - Is PassengerModule a valid NestJS module?
bmn-passenger  | - If OpenTelemetryOptions is a provider, is it part of the current PassengerModule?
bmn-passenger  | - If OpenTelemetryOptions is exported from a separate @Module, is that module imported within PassengerModule?
bmn-passenger  |   @Module({
bmn-passenger  |     imports: [ /* the Module containing OpenTelemetryOptions */ ]
bmn-passenger  |   })
bmn-passenger  |
bmn-passenger  |     at Injector.lookupComponentInParentModules (/usr/src/app/passenger/node_modules/@nestjs/core/injector/injector.js:254:19)
bmn-passenger  |     at async Injector.resolveComponentInstance (/usr/src/app/passenger/node_modules/@nestjs/core/injector/injector.js:207:33)
bmn-passenger  |     at async resolveParam (/usr/src/app/passenger/node_modules/@nestjs/core/injector/injector.js:128:38)
bmn-passenger  |     at async Promise.all (index 0)
bmn-passenger  |     at async Injector.resolveConstructorParams (/usr/src/app/passenger/node_modules/@nestjs/core/injector/injector.js:143:27)
bmn-passenger  |     at async Injector.loadInstance (/usr/src/app/passenger/node_modules/@nestjs/core/injector/injector.js:70:13)
bmn-passenger  |     at async Injector.loadProvider (/usr/src/app/passenger/node_modules/@nestjs/core/injector/injector.js:97:9)
bmn-passenger  |     at async /usr/src/app/passenger/node_modules/@nestjs/core/injector/instance-loader.js:56:13
bmn-passenger  |     at async Promise.all (index 6)
bmn-passenger  |     at async InstanceLoader.createInstancesOfProviders (/usr/src/app/passenger/node_modules/@nestjs/core/injector/instance-loader.js:55:9)
```

### Issues with `@nestjs` packages versions
```js
node_modules/@nestjs/core/adapters/http-adapter.d.ts:28:5 - error TS2416: Property 'search' in type 'AbstractHttpAdapter<TServer, TRequest, TResponse>' is not assignable to the same property in base type 'HttpServer<TRequest, TResponse, any>'.
  Type '{ (port: string | number, callback?: () => void): any; (port: string | number, hostname: string, callback?: () => void): any; }' is not assignable to type '{ (handler: RequestHandler<TRequest, TResponse>): any; (path: string, handler: RequestHandler<TRequest, TResponse>): any; }'.
    Types of parameters 'port' and 'handler' are incompatible.
      Type 'RequestHandler<TRequest, TResponse>' is not assignable to type 'string | number'.

28     search(port: string | number, callback?: () => void): any;
       ~~~~~~
node_modules/@nestjs/core/adapters/http-adapter.d.ts:29:5 - error TS2416: Property 'search' in type 'AbstractHttpAdapter<TServer, TRequest, TResponse>' is not assignable to the same property in base type 'HttpServer<TRequest, TResponse, any>'.
  Type '{ (port: string | number, callback?: () => void): any; (port: string | number, hostname: string, callback?: () => void): any; }' is not assignable to type '{ (handler: RequestHandler<TRequest, TResponse>): any; (path: string, handler: RequestHandler<TRequest, TResponse>): any; }'.

29     search(port: string | number, hostname: string, callback?: () => void): any;
       ~~~~~~
node_modules/@nestjs/core/nest-application.d.ts:28:5 - error TS2416: Property 'getHttpAdapter' in type 'NestApplication' is not assignable to the same property in base type 'INestApplication<any>'.
  Type '() => AbstractHttpAdapter<any, any, any>' is not assignable to type '() => HttpServer<any, any, any>'.
    Call signature return types 'AbstractHttpAdapter<any, any, any>' and 'HttpServer<any, any, any>' are incompatible.
      The types of 'search' are incompatible between these types.
        Type '{ (port: string | number, callback?: () => void): any; (port: string | number, hostname: string, callback?: () => void): any; }' is not assignable to type '{ (handler: RequestHandler<any, any>): any; (path: string, handler: RequestHandler<any, any>): any; }'.
          Types of parameters 'port' and 'handler' are incompatible.
            Type 'RequestHandler<any, any>' is not assignable to type 'string | number'.

28     getHttpAdapter(): AbstractHttpAdapter;
```

### Missing `ts-node` tool/package
```js
[nodemon] starting `ts-node src/main.ts`
sh: ts-node: not found
[nodemon] failed to start process, "ts-node" exec not found
```

And a small refactor on `.env` files/config to normalize them (`POSTGRES_Database` and `RABBITMQ_Host` when everything else was full uppercase)

**No more issues when running `npm run build`** 🥳

### CODE REVIEW

File: src/building-blocks/openTelemetry/open-telemetry.module.ts [hunk 1]
```diff
@@ -6,7 +6,7 @@ import { OpenTelemetryOptions, OpenTelemetryTracer} from './open-telemetry-trace
     provide: 'IOpenTelemetryTracer',
     useClass: OpenTelemetryTracer
   }],
-  exports: ['IOpenTelemetryTracer']
+  exports: ['IOpenTelemetryTracer', OpenTelemetryOptions]
```
Comment (meysamhadeli) [inline src/building-blocks/openTelemetry/open-telemetry.module.ts:9]: Why did you expose `OpenTelemetryOptions` here? we don't need to expose `OpenTelemetryOptions` to outside. we only need `IOpenTelemetryTracer` to use tracer.

**Review by meysamhadeli — COMMENTED** [review]
Thanks for contribution :)
Please check some comments.

**Comment by cyrusssschen** [comment]
@TBG-FR I got the same error when trying to build this project, but all good after this PR is applied.

**Comment by meysamhadeli** [comment]
Because i didn't get any comments for this PR I updated all packages to last version and also in build every things is ok in main. 
So I closed this PR and if any problem is exist please create separated PR for that.

Thanks :)

**Pull request closed without merging**